File: leetcode/01572_matrix-diagonal-sum/1572-matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        ll = len(mat)
        half = ll // 2
        j = ll-1
        total = 0
        for i in range(ll):
            if i == half and ll % 2 == 1:
                total += mat[i][i]
                j -= 1
            else:
                total += mat[i][i] + mat[i][j]
                i += 1
                j -= 1
        return total

File: leetcode/01572_matrix-diagonal-sum/1572-matrix-diagonal-sum.js
/**
 * @param {number[][]} mat
 * @return {number}
 */
var diagonalSum = function (mat) {
    // 00 02
    // 11 11
    // 22 20
    let start = 0,
        end = mat.length - 1,
        mid = parseInt(mat.length / 2);
    sum = 0;
    for (let i = 0; i < mat.length; i++) {
        //console.log(mat[i][start], mat[i][end]);
        sum += mat[i][start] + mat[i][end];
        start++;
        end--;
        //console.log(sum);
    }
    if (mat.length % 2 != 0) {
        sum -= mat[mid][mid];
    }
    return sum;
};